src/AWS.elm

```diff
@@ -4,31 +4,89 @@ module AWS exposing (..)
 
 __Experimental: Work in progress__
 
-@docs Credentials, ServiceConfig
+@docs ServiceConfig, Credentials, credentials, AccessKeyId, SecretAccessKey, SessionToken
+
+## Request Signing
+
+@docs Request, SignedRequest, signV4
 
 -}
 
+import AWS.Config
+import AWS.Signers.V4 exposing (sign)
+import AWS.Http
+import Date exposing (Date)
+import Http
+
 
 {-| Credentials for accessing AWS services.
 -}
-type alias Credentials =
-    { accessKeyId : String
-    , secretAccessKey : String
-    , sessionToken : Maybe String
-    }
+type Credentials
+    = Credentials AWS.Config.Credentials
```

Comment:
Hiding the details of these types. The `AWS.Config.Credentials` is not exposed, so it will be impossible to construct directly. Then we can update the implementation details without having to do a major release